src/createAsyncStoragePersistor-experimental/index.ts

```diff
@@ -0,0 +1,78 @@
+interface AsyncStorage {
+  getItem: (key: string) => Promise<string>
+  setItem: (key: string, value: string) => Promise<unknown>
+  removeItem: (key: string) => Promise<unknown>
+}
+
+interface CreateAsyncStoragePersistorOptions {
+  /** The storage client used for setting an retrieving items from cache */
+  storage: AsyncStorage
+  /** The key to use when storing the cache */
+  key?: string
+  /** To avoid spamming,
+   * pass a time in ms to throttle saving the cache to disk */
+  throttleTime?: number
+}
+
+export const asyncStoragePersistor = ({
+  storage,
+  key = `REACT_QUERY_OFFLINE_CACHE`,
+  throttleTime = 1000,
+}: CreateAsyncStoragePersistorOptions) => {
+  return {
+    persistClient: asyncThrottle(
+      persistedClient => storage.setItem(key, JSON.stringify(persistedClient)),
+      { interval: throttleTime }
+    ),
+    restoreClient: async () => {
+      const cacheString = await storage.getItem(key)
+
+      if (!cacheString) {
+        return
+      }
+
+      return JSON.parse(cacheString)
+    },
+    removeClient: () => storage.removeItem(key),
+  }
+}
+
+function asyncThrottle<T>(
```

Comment:
```typescript
   const timeout = (ms: number) => new Promise(resolve => setTimeout(resolve, ms))

    persistClient: async persistedClient => {
      await timeout(throttleTime)
      await storage.setItem(key, JSON.stringify(persistedClient))
    },
```

Can we do something as simple as this? Isn't the time-based invalidation already handled in `persistQueryClient` already?

Reply:
We want to queue calls and therefore we will cached data at some point otherwise we will call `setItem` too often